src/engine.rs

```diff
@@ -1066,4 +1066,99 @@ mod tests {
         drop(engine);
         RaftLogEngine::open(cfg).unwrap();
     }
+
+    #[test]
+    fn test_parallel_recover_merge() {
```

Comment:
Could you verify the `GlobalStats` after the merge too? I previously spotted a bug with it and fixed it.